config/routes.rb

```diff
@@ -1,17 +1,19 @@
 Spree::Core::Engine.routes.draw do
   namespace :admin do
-    resources :blog_entries
+    resources :blogs do
+      resources :blog_entries
+    end
   end
 
   resources :blog_entries
+  get '/:blog_id/:slug' => 'blog_entries#show', as: :blog_entry_permalink
```

Comment:
I've updated this, in my latest code. I think I need to update the pull-request. There is an index on [:blog_id, :permalink] at the schema level.